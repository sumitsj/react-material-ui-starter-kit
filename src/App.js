import React from 'react';
import { HashRouter as Router, Route } from 'react-router-dom';
import Container from '@material-ui/core/Container';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
import Home from './pages/Home';
import Login from './pages/Login';
import StateProvider, { initialState, reducer } from './store';
import PrivateRoute from './components/PrivateRoute';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#4489ff',
    },
  },
});

function App() {
  return (
    <ThemeProvider theme={theme}>
      <StateProvider initialState={initialState} reducer={reducer}>
        <Container maxWidth="lg" style={{ marginTop: 90 }}>
          <Router>
            <PrivateRoute path="/" exact component={Home} />
            <Route path="/login" component={Login} />
          </Router>
        </Container>
      </StateProvider>
    </ThemeProvider>
  );
}

export default App;
