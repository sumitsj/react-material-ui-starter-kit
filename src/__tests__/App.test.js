import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe('App', () => {
  beforeAll(() => {
    localStorage.getItem.mockImplementation(() => 'true');
  });

  afterEach(() => {
    localStorage.getItem.mockRestore();
  });

  describe('HTML Structure', () => {
    it('should render properly', () => {
      const { asFragment } = render(<App />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
