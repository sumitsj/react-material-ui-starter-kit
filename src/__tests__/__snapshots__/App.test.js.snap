// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App HTML Structure should render properly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-logo-1 MuiTypography-h6"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-logoLink-2 MuiTypography-colorPrimary"
          href="/"
        >
          Assisted HIP
        </a>
      </h6>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
    style="margin-top: 90px;"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6"
      style="background-color: cyan;"
    >
      Home Page
    </h6>
  </div>
</DocumentFragment>
`;
